distribution/docker/src/docker/Dockerfile

```diff
@@ -16,10 +16,10 @@ FROM centos:7 AS builder
 ENV PATH /usr/share/elasticsearch/bin:$PATH
 ENV JAVA_HOME /opt/jdk-${jdkVersion}
 
-RUN curl -L --retry 8 -s -S ${jdkUrl} --continue-at - --output /tmp/jdk.tar.gz && \
-    tar -C /opt -zxf /tmp/jdk.tar.gz && \
-    rm -Rf /tmp/jdk.tar.gz
-
+RUN for iter in {1..10}; do curl -L -s -S ${jdkUrl} | tar -C /opt -zx && \
+    exit_code=0 && break || exit_code=\$? && echo "download error: retry $iter in 10s" && sleep 10; done; \
+    (exit $exit_code)
+    
```

Comment:
(super) nit: we could remove the 4 space characters in line 22 (so that there's just a newline).